File: BniAposApp/app/src/main/java/com/example/bniapos/BniApplication.java
package com.example.bniapos;

import android.app.Application;
import android.content.Context;

import com.example.paymentsdk.CTIApplication;


public class BniApplication extends Application {

    /**
     * Create.
     */
    @Override
    public void onCreate() {
        super.onCreate();
        CTIApplication.setContext(getApplicationContext());
        bindSdkDeviceService(getApplicationContext());
    }


    @Override
    public void onTerminate() {
        super.onTerminate();
        CTIApplication.unbindServiceConnection();
        System.exit(0);
    }


    private static void bindSdkDeviceService(Context context) {
        CTIApplication.bindSdkDeviceService(context);
    }


}

File: BniAposApp/app/src/main/java/com/example/bniapos/BniApplication.kt
package com.example.bniapos

import android.app.Application
import android.content.Context
import com.example.paymentsdk.CTIApplication
import kotlin.system.exitProcess

class BniApplication : Application() {


    /**
     * Create.
     */
    override fun onCreate() {
        super.onCreate()
        context = applicationContext
        CTIApplication.setContext(applicationContext)
        bindSdkDeviceService(applicationContext)
    }

    override fun onTerminate() {
        super.onTerminate()
        CTIApplication.unbindServiceConnection()
        exitProcess(0)
    }

    companion object {
        var context: Context? = null

        private fun bindSdkDeviceService(context: Context) {
            CTIApplication.bindSdkDeviceService(context)
        }
    }

}